core/src/test/java/org/elasticsearch/common/geo/builders/PolygonBuilderTests.java

```diff
@@ -124,4 +125,24 @@ public void testCoerceHole() {
         assertThat("hole should have been closed via coerce", pb.holes().get(0).coordinates(false).length, equalTo(4));
     }
 
+    public void testHoleThatIsSouthOfPolygon() {
+        try {
+            PolygonBuilder pb = new PolygonBuilder(new CoordinatesBuilder().coordinate(4, 3).coordinate(3, 2).coordinate(3, 3).close());
+            pb.hole(new LineStringBuilder(new CoordinatesBuilder().coordinate(4, 2).coordinate(3, 1).coordinate(4, 1).close()));
+            pb.build();
+        } catch (InvalidShapeException e) {
+            assertEquals("Hole lies outside shell at or near point (4.0, 1.0, NaN)", e.getMessage());
```

Comment:
nit: there is the neat expectThrows() utility in LuceneTestCase that makes test for exceptions a little more succinct. MIght help here (and in the other test) as well

Reply:
D'oh, I forgot about that. I pushed dc88dace9b